File: C#/SecuJournal/sjSettings.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace SecuJournal
{
    public partial class sjSettings : Form
    {
        public sjSettings()
        {
            InitializeComponent();
        }

        private void sjSettings_Load(object sender, EventArgs e)
        {
            // Load default save directory settings.
            DSLText.Text = Properties.Settings.Default.DefaultSaveDirectory;

            // Load word-wrap settings.
            if (Properties.Settings.Default.WordWrap == false)
            {
                WWCheckbox.Checked = false;
            }
            else
            {
                WWCheckbox.Checked = true;
            }

            // Load encryption settings.
            if (Properties.Settings.Default.OTFEncryption == false)
            {
                OTFECheckBox.Checked = false;
                OTFEPassword.Enabled = false;
            }
            else
            {
                OTFECheckBox.Checked = true;
                OTFEPassword.Enabled = true;
            }

            // User set password is always the OTFE password when enabled.
            OTFEPassword.Text = Properties.Settings.Default.SJPassword;
        }

        private void sjSettings_FormClosing(object sender, FormClosingEventArgs e)
        {
            // Need to use a new sjEntry for sjEntry properties because sjEntry = sjEntry all the time.
            sjEntry sje = new sjEntry();

            // Save all user settings.
            if (OTFEPassword.Text == "")
            {

            }
            else
            {
                sje.EncryptTextToolStripMenuItem.Enabled = true;
                sje.DecryptTextToolStripMenuItem.Enabled = true;
            }

            Properties.Settings.Default.SJPassword = OTFEPassword.Text;
            Properties.Settings.Default.Save();
        }

        private void DSLBrowseButton_Click(object sender, EventArgs e)
        {
            FolderBrowserDialog folderDialogue = new FolderBrowserDialog();
            folderDialogue.ShowNewFolderButton = true;
            if (folderDialogue.ShowDialog() == DialogResult.OK)
            {
                DSLText.Text = folderDialogue.SelectedPath;
            }
        }

        private void WWCheckbox_CheckedChanged(object sender, EventArgs e)
        {
            if (WWCheckbox.Checked == true)
            {
                Properties.Settings.Default.WordWrap = true;
            }
            else
            {
                Properties.Settings.Default.WordWrap = false;
            }
        }

        private void WTCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            if (WTCheckBox.Checked == true)
            {
                Properties.Settings.Default.WriteTitle = true;
            }
            else
            {
                Properties.Settings.Default.WriteTitle = false;
            }
        }

        private void OTFECheckBox_CheckedChanged(object sender, EventArgs e)
        {
            if (WTCheckBox.Checked == true)
            {
                Properties.Settings.Default.WriteTitle = true;
            }
            else
            {
                Properties.Settings.Default.WriteTitle = false;
            }
        }
    }
}

File: SecuJournal/SJSettings.vb
Public Class SJSettings
    Private Sub SJSettings_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        'Load default save directory settings.
        DSLText.Text = My.Settings.DefaultSaveDirectory

        'Load word wrap settings.
        If My.Settings.WordWrap = 0 Then
            WWCheckbox.Checked = False
        Else
            WWCheckbox.Checked = True
        End If

        'Load write title settings.
        If My.Settings.WriteTitle = 0 Then
            WTCheckBox.Checked = False
        Else
            WTCheckBox.Checked = True
        End If

        'Load encryption settings.
        If My.Settings.OTFEncryption = 0 Then
            OTFECheckBox.Checked = False
            OTFEPassword.Enabled = False
        Else
            OTFECheckBox.Checked = True
            OTFEPassword.Enabled = True
        End If

        OTFEPassword.Text = My.Settings.SJPassword
    End Sub

    Private Sub SJSettings_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        My.Settings.DefaultSaveDirectory = DSLText.Text
        If OTFEPassword.Text = "" Then
        Else
            SJEntry.EncryptTextToolStripMenuItem.Enabled = True
            SJEntry.DecryptTextToolStripMenuItem.Enabled = True
        End If
        My.Settings.SJPassword = OTFEPassword.Text
        My.Settings.Save()
    End Sub

    Private Sub DSLBrowseButton_Click(sender As Object, e As EventArgs) Handles DSLBrowseButton.Click
        Dim FolderDialogue As New FolderBrowserDialog
        FolderDialogue.ShowNewFolderButton = True
        If (FolderDialogue.ShowDialog() = DialogResult.OK) Then
            DSLText.Text = FolderDialogue.SelectedPath
        End If
    End Sub

    Private Sub OTFECheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles OTFECheckBox.CheckedChanged
        If OTFECheckBox.Checked = True Then
            My.Settings.OTFEncryption = 1
            OTFEPassword.Enabled = True
        Else
            My.Settings.OTFEncryption = 0
            OTFEPassword.Enabled = False
        End If
    End Sub

    Private Sub WWCheckbox_CheckedChanged(sender As Object, e As EventArgs) Handles WWCheckbox.CheckedChanged
        If WWCheckbox.Checked = True Then
            My.Settings.wordwrap = 1
        Else
            My.Settings.wordwrap = 0
        End If
    End Sub

    Private Sub WTCheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles WTCheckBox.CheckedChanged
        If WTCheckBox.Checked = True Then
            My.Settings.WriteTitle = 1
        Else
            My.Settings.WriteTitle = 0
        End If
    End Sub
End Class